server/routes/journeys/routes.ts

```diff
@@ -24,14 +24,27 @@ export const JourneyRoutes = (services: Services) => {
   router.use('/', EditLogCodeRoutes({ services, path: '/edit-log-code' }))
   router.use('/', UpdateInvestigationRoutes({ services, path: '/update-investigation' }))
 
+  /* eslint-disable no-param-reassign */
+  const mergeObjects = <T extends Record<string, unknown>>(destination: T, source: Partial<T>) => {
+    Object.entries(source).forEach(([key, value]) => {
+      if (typeof value === 'object' && !Array.isArray(value)) {
+        if (!destination[key]) {
+          // @ts-expect-error set up object for future recursive writes
+          destination[key] = {}
+        }
+        mergeObjects(destination[key] as Record<string, unknown>, value)
+      } else {
+        // @ts-expect-error unexpected types
+        destination[key] = value
+      }
+    })
+  }
+
```

Comment:
looks like it belongs to an util class.
should we import some library such as lodash to handle deep merge?

Reply:
I agree with Bill in that this should either go in a util class or just be removed as a function and move the code inside the `if (process.env.NODE_ENV === 'e2e-test') {`

Reply:
I actually prefer just moving the code inside the `if`. If we are having to do a deep merge in more than one place, I think I would prefer to use a library for it.

Reply:
Moved the declaration to be inside the `if`. it is a recursive function though so it needs to stay as a function

Reply:
Ah of course :)